### PR TITLE
Add PropertiesAutowired to enable property injection

### PR DESCRIPTION
**Pull request by tabman83**
This is needed to enable property injection.

### CODE REVIEW

**Comment by chomick** [comment]
Nino, please follow the branch naming convention

File: src/Eshopworld.Data.CosmosDb/CosmosDbModule.cs [hunk 1]
```diff
@@ -18,7 +18,8 @@ protected override void Load(ContainerBuilder builder)
                 .SingleInstance();
 
             builder.RegisterType<CosmosDbRepository>()
-                .As<ICosmosDbRepository>();
+                .As<ICosmosDbRepository>()
+                .PropertiesAutowired();
```
Comment (chomick) [inline src/Eshopworld.Data.CosmosDb/CosmosDbModule.cs:22]: Why do we do that?

Comment (chomick) [reply src/Eshopworld.Data.CosmosDb/CosmosDbModule.cs:22]: The reason we did the settings on Property not in ctor was that we did not have the instance available in container and we were expecting someone to provide it after initialisation (via property).
Or am I missing something?

Comment (tabman83) [reply src/Eshopworld.Data.CosmosDb/CosmosDbModule.cs:22]: That's correct but the property won't be injected if we don't have this, or at least this is my understanding of AutoFac

**Comment by tabman83** [comment]
I have verified through Captain Hook that this is needed in order to correctly inject the property.

**Review by mszymeshop — APPROVED** [review]

**Review by dawidkeshop — APPROVED** [review]

**Comment by tabman83** [comment]
Closing as I will be using constructor injection approach.

**Pull request closed without merging**